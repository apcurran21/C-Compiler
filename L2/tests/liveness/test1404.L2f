(@encodeNumber
	1
	%n <- rdi
	%encodedN <- %n
	%encodedN <<= 1
	%encodedN += 1
	rax <- %encodedN
	return
)
