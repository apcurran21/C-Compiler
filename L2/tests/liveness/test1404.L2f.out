(
(in
(r12 r13 r14 r15 rbp rbx rdi)
(%n r12 r13 r14 r15 rbp rbx)
(%encodedN r12 r13 r14 r15 rbp rbx)
(%encodedN r12 r13 r14 r15 rbp rbx)
(%encodedN r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(%n r12 r13 r14 r15 rbp rbx)
(%encodedN r12 r13 r14 r15 rbp rbx)
(%encodedN r12 r13 r14 r15 rbp rbx)
(%encodedN r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
